### ISSUE TITLE
me again

### ISSUE THREAD

**Issue by lisanka93** [adeshpande3/Facebook-Messenger-Bot#3]
thanks for your help last time!
my model is finally done training (yeay) but not i ran into another error

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [192] rhs shape= [448]
	 [[Node: save/Assign_1 = Assign[T=DT_FLOAT, _class=["loc:@embedding_rnn_seq2seq/embedding_rnn_decoder/rnn_decoder/output_projection_wrapper/basic_lstm_cell/bias"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/cpu:0"](embedding_rnn_seq2seq/embedding_rnn_decoder/rnn_decoder/output_projection_wrapper/basic_lstm_cell/bias, save/RestoreV2_1)]]

but I dont understand where the tensors are defined :/
dont I need to feed tf.train.Saver() some variables?

**Comment by adeshpande3** [comment]
So, you've trained your model and you have it saved in your models folder, correct? If so, double check that you have a file structure like this. 

<img width="639" alt="models" src="https://user-images.githubusercontent.com/13543909/29324034-aa672562-8197-11e7-81bd-a7452a373d0a.png">

 Are you getting this error when you try to run saver.restore(sess, tf.train.latest_checkpoint('models/'))?

**Comment by lisanka93** [comment]
yes file structure is correct :)
and yes I get the error if i run saver.restore etc.

**Comment by adeshpande3** [comment]
Okay so that error normally occurs when the graph you save is different from the graph you've defined. Double check to make sure that the shapes of all your variables are the same as the graph you saved, same number of LSTM units, etc. From your error, I'm fairly certain it has something to do with the number of LSTM units. Try increasing/decreasing that number to figure out if "lhs shape=[192] rhs shape=[448]" changes.

Also, when you run the saver.restore function, you *don't* have to run sess.run(tf.global_variables_initializer()) since the values for the variables are being loaded in. Try removing that.

**Comment by lisanka93** [comment]
THANK YOU!!! yes it was that!

**Comment by lisanka93** [comment]
and my last question is
if the chatbot doesnt have a response (list of responses is 0) is there a change to make it learn those?

**Comment by adeshpande3** [comment]
By list of responses, do you mean that the call to idsToSentence always returns a list thats empty? What do you get when you just print the ids variable? Is it just padding and EOS tokens?

**Comment by lisanka93** [comment]
so it works but once in a while it crashes and I assume that for this statement the model doesnt have a reply

**Comment by adeshpande3** [comment]
Yeah I would assume that's the issue. With problems like these, the issue could stem from a number of different places (not enough training data, too small/big of a model, not training enough, wrong hyperparameters, etc) so it's tough to really tell what a fix is.

To try to improve the performance, a useful exercise could be to try training the model on a smaller subset of the training data (lets say 100 or so conversations) and then training the model and making sure that the bot is able to respond correctly to just those convos. If it's not able to learn it, then you know that there's probably some problem with your model architecture, hyperparameters, etc
